modules/internal/ChapelBase.chpl

```diff
@@ -1984,6 +1984,9 @@ module ChapelBase {
   pragma "task complete impl fn"
   extern proc chpl_comm_task_end(): void;
 
+  // TODO do we need "task complete impl fn" pragma?
```

Comment:
@ronawho -- Do you know offhand what should I do here?  I can dig into the compiler unless you have an answer.

Reply:
I see. That's interesting, unordered optimization fires early in LICM, that's before any GPU operation. So, I don't know what happens when it fires for a gpu-eligible loop. Potentially:

- we replace the assignment primitive with the optimized one
- it's lowered back to assignment during codegen because the GPU kernel is fully local.

In any case, the lack or existence of this pragma could impact the optimization. I'll take a look.

Reply:
It looks like adding this pragma is safer -- there are already multiple functions with this pragma on. Unordered optimization seems to collect all such functions. I remain doubtful it matters for any real purpose today, but it does look like the right thing to do.